### ISSUE TITLE
Need additional station connect parameters

### ISSUE THREAD

**Issue by negativekelvin** [espressif/esp32-wifi-lib#4]
It takes 1s longer to boot and connect to an ap on channel 11 than on channel 1.  After the first time we connect to an ap, we can remember its channel.

So we need to be able to provide a prioritized list of channels to use when scanning in sta mode.  Also we should be able to specify channel dwell times.  It is assumed that directed probe request is used here.

Also related: https://github.com/espressif/esp-idf/issues/326

**Comment by negativekelvin** [comment]
@TimXia any info on this topic?

**Comment by TimXia** [comment]
@negativekelvin I guess that you mean fast connect. We will try to implement what you say.

**Comment by TimXia** [comment]
@negativekelvin This feature will be implemented in IDFv2.1.

**Comment by negativekelvin** [comment]
@TimXia Thanks for adding these features.  

Regarding  

> If AP is not on the specified channel, continue scanning the next
> channel

I think there is a case when we don't want to do that, APSTA mode so softAP can still operate on the channel.

**Comment by TimXia** [comment]
@negativekelvin The channel of softAP follows that of STA. So, I think it does not impact softAP too much if STA continues scanning the next channel.

**Comment by negativekelvin** [comment]
@TimXia yes but if there are client connected to softap we don't want to change channel on them.  Also if we know for certain AP channel we may not want to use battery to scan other channel.

**Comment by negativekelvin** [comment]
@TimXia also if we look at some [data](https://wigle.net/stats#octetstats) we see that most ap on channel 1,6,11 so we want to be able to scan these channels first before other channels.
